10-JavaScript/closures-continued.md

```diff
@@ -4,48 +4,110 @@
 
 Often when building a single page application, you will need to have many UI elements that each do something similar. An example of this is a list of contacts, where clicking on the name of a contact pops up a modal showing details.
 
-A simpler example is a set of buttons which can each change the color of a box on the screen, [as in this codepen](https://codepen.io/adadev/pen/ZmbMdq). Don't worry too much about the code itself yet, just the functionality of the buttons.
+A simpler example is a set of buttons which can each change the color of a box on the screen, [as in this codesandbox](https://codesandbox.io/s/kind-ritchie-6z3dz). Don't worry too much about the code itself yet, just the functionality of the buttons.
 
 Because the UI elements have such similar behavior, we want to use the same function to handle each one. However, that function needs to know a little bit of extra information about which element it's attached to. This is where closures come in.
 
 **A closure is a way to attach some extra information to a function.**
 
-## Closures with jQuery
 
-Now that we've explored the concepts behind closures a bit, let's return to our [original example with the buttons](https://codepen.io/adadev/pen/ZmbMdq).
+Doing event handling in the browser is something new and we will handle it a bit differently in React.  However these are common use-cases where people take advantage of closures when interacting with the DOM.
+
+## Closures with vanilla (plain) JavaScript
+
+Now that we've explored the concepts behind closures a bit, let's return to our [original example with the buttons](https://codesandbox.io/s/kind-ritchie-6z3dz).
 
 With your neighbors, look at the JavaScript code in that pen, open your Dev Tools to see the console, and answer the following questions:
 
+- Where is the closure created?
 - What is the outer function?
 - What is the inner function?
 - What variable from the outer function is being referenced in the inner function?
 - How many different click handlers are created by this code?
 
+## JavaScript in the browser
+
+### Where the code starts running
+
+The code below tells Javascript to call `mainFunction` after the HTML page has finished loading.
+
+```javascript
+// Run the main function only after the HML has loaded
+if (document.readyState === "loading") {
+  // Loading hasn't finished yet
+  document.addEventListener("DOMContentLoaded", mainFunction);
+} else {
+  // `DOMContentLoaded` has already fired
+  mainFunction();
+}
+```
+
+### The mainFunction
```

Comment:
I'm wondering if there's a naming convention for a different name besides `mainFunction`? If there isn't or if this is the normal thing, then that's totally fine. Also, I couldn't think of other suggestions anyway, besides something like `onLoad` but I don't know if that's confusing.